src/enzo/Grid_ComputeAccelerationFieldExternal.C

```diff
@@ -408,57 +414,58 @@ int grid::ComputeAccelerationFieldExternal()
 
             rsquared = xpos*xpos + ypos*ypos + zpos*zpos;
 
-            double accelsph, accelcylR, accelcylz, zheight, xpos1, ypos1, zpos1;
+            double accelsph, accelcylR, accelcylz, zheight, xdisk, ydisk, zdisk;
 
             /* Compute z and r_perp (AngularMomentum is angular momentum 
              * and must have unit length). */
 
             /* magnitude of z = r.L in L direction */
 
-            zheight=AngularMomentumx*xpos + AngularMomentumy*ypos + AngularMomentumz*zpos;
-
-            /* position in plane of disk */
-
-            xpos1=xpos-zheight*AngularMomentumx;
-            ypos1=ypos-zheight*AngularMomentumy;
-            zpos1=zpos-zheight*AngularMomentumz;
-
-            radius = sqrt(xpos1*xpos1 + ypos1*ypos1 + zpos1*zpos1 + zheight*zheight);
-            rcyl = sqrt(xpos1*xpos1 + ypos1*ypos1 + zpos1*zpos1);
-            radius = radius*LengthUnits;
-            rcyl = rcyl*LengthUnits;
-            accelsph = (GravConst)*MBulge*SolarMass/POW(radius+rBulge*Mpc_cm,2)
-                     + pi*GravConst*densDMConst*POW(rDMConst*Mpc_cm,3)/POW(radius,2)
-                       *(-2.0*atan(radius/rDMConst/Mpc_cm)
-                         +2.0*log(1.0+radius/rDMConst/Mpc_cm)
-                         +log(1.0+POW(radius/rDMConst/Mpc_cm,2))
-                        );
-            accelcylR = GravConst*MSDisk*SolarMass*rcyl/sqrt(POW(POW(rcyl,2)
-                      + POW(SDiskScaleHeightR*Mpc_cm+sqrt(POW(zheight*LengthUnits,2)
-                      + POW(SDiskScaleHeightz*Mpc_cm,2)),2),3));
-            accelcylz = GravConst*MSDisk*SolarMass/sqrt(POW(zheight*LengthUnits,2)
-                      + POW(SDiskScaleHeightz*Mpc_cm,2))*zheight*LengthUnits/sqrt(POW(POW(rcyl,2)
-                      + POW(SDiskScaleHeightR*Mpc_cm+sqrt(POW(zheight*LengthUnits,2)
-                      + POW(SDiskScaleHeightz*Mpc_cm,2)),2),3))
-                        *(  SDiskScaleHeightR*Mpc_cm+sqrt(POW(zheight*LengthUnits,2)
-                          + POW(SDiskScaleHeightz*Mpc_cm,2))
-                         )/AccelUnits;
+            zheight = AngularMomentumx*xpos + AngularMomentumy*ypos + AngularMomentumz*zpos;
+
+            /* position in plane of disk in code units */
+
+            xdisk=xpos-zheight*AngularMomentumx;
+            ydisk=ypos-zheight*AngularMomentumy;
+            zdisk=zpos-zheight*AngularMomentumz;
+
+            // Put zheight into cgs for easier calculation of acceleration
+            zheight *= LengthUnits;
+
+            radius = sqrt(rsquared) * LengthUnits;
+            rcyl = sqrt(xdisk*xdisk + ydisk*ydisk + zdisk*zdisk) * LengthUnits;
+            
+            accelsph = GravConst * 4.0*pi*rho_0*POW(Rs,3.0)
+                        *(log((Rs+radius)/Rs) - radius/(Rs+radius));
+            accelsph += (GravConst)*MBulge/POW(radius+rBulge,2);
+
+            accelcylR = GravConst*MSDisk*rcyl/sqrt(POW(POW(rcyl,2) // was rcyl converted to cgs? Yes; line 433
```

Comment:
the comment seems to be inaccurate; it looks like it's now line 436 (according to GitHub at least).  Maybe just say "rcyl already converted to CGS"